include/dicom/dicom.h

```diff
@@ -220,1618 +272,586 @@ typedef struct _DcmError DcmError;
+ * :return: true on success
  */
 DCM_EXTERN
-const char *dcm_element_get_value_OD(const DcmElement *element);
+bool dcm_element_get_value_integer(DcmError **error,
+                                   const DcmElement *element,
+                                   uint32_t index,
+                                   int64_t *value);
 
 /**
- * Get value of a Data Element with Value Representation OF (Other Float).
+ * Set the value of an element to an integer. The element must have an
+ * appropriate tag.
  *
+ * :param error: Error structure pointer
  * :param element: Pointer to Data Element
+ * :param value: Integer value
  *
- * :return: Pointer to memory location where value is stored
+ * :return: true on success
  */
 DCM_EXTERN
-const char *dcm_element_get_value_OF(const DcmElement *element);
+bool dcm_element_set_value_integer(DcmError **error,
+                                   DcmElement *element,
+                                   int64_t value);
 
 /**
- * Get value of a Data Element with Value Representation OL (Other Long).
+ * Set the value of an element to an array of numeric values. The element must
+ * have an appropriate tag.
+ *
+ * Although the value passed is `int*`, it should
+ * be a pointer to an array of 16- to 64-bit numeric values of the
+ * appropriate type for the element VR.
  *
+ * On success, if `steal` is true, ownership of `values` passes to
+ * `element`, ie. it will be freed when `element` is destroyed. If `steal` is
+ * false, then a copy is made of `values` and ownership is not transferred.
+ *
+ * :param error: Error structure pointer
  * :param element: Pointer to Data Element
+ * :param values: Array of values
+ * :param vm: Number of values
+ * :param steal: if true, ownership of values passes to element
  *
- * :return: Pointer to memory location where value is stored
+ * :return: true on success
  */
 DCM_EXTERN
-const char *dcm_element_get_value_OL(const DcmElement *element);
+bool dcm_element_set_value_numeric_multi(DcmError **error,
+                                         DcmElement *element,
+                                         int *values,
+                                         uint32_t vm,
+                                         bool steal);
 
 /**
- * Get value of a Data Element with Value Representation OV (Other Very Long).
+ * Get a double from a 32- or 64-bit float element.
  *
+ * :param error: Error structure pointer
  * :param element: Pointer to Data Element
+ * :param index: Zero-based index of value within the Data Element
+ * :param value: Integer value return
  *
- * :return: Pointer to memory location where value is stored
+ * :return: true on success
  */
 DCM_EXTERN
-const char *dcm_element_get_value_OV(const DcmElement *element);
+bool dcm_element_get_value_double(DcmError **error,
+                                  const DcmElement *element,
+                                  uint32_t index,
+                                  double *value);
```

Comment:
I looked into this, and `float32_t` and `float64_t` are not in the C standard yet. I think we should probably avoid making extra typedefs and just stick with `float` and `double`.

Reply:
Hmm...that's a good point. How strict do we want (have) to be about that? From a design perspective, I would prefer the exact types. What are your concerns about extra typedefs?

Reply:
It's extra mental work for our users (they will need to look up and then remember what `DcmFloat32` and `DcmFloat64` mean exactly), they will probably lose some degree of syntax highlighting, it will look inconsistent next to the other numeric types, it's extra code for us to maintain, and I think it's trying to fix a problem that doesn't really exist (all platforms, even bizarre ones like emscripten, use IEEE floating point, so float == 32 and double == 64).